### PR TITLE
Add link title support for commonmark

### LINKED ISSUE #67: Link titles not supported?
[Example 481](https://spec.commonmark.org/0.30/#example-481 "title goes here") of the commonmark spec shows the use of a title in a link:

```markdown
[link](/uri "title")
```

however I don't see the title property in the [Link](https://apple.github.io/swift-markdown/documentation/markdown/link) documentation

When I test, it seems as though the title is discarded:
```swift
let document = Document(parsing: "[link](/uri \"title\")")
print(document.debugDescription())
```

Outputs:
```
Document
└─ Paragraph
   └─ Link destination: "/uri"
      └─ Text "link"
```

Title is supported for Images:
```swift
let document2 = Document(parsing: "![foo](/url \"title\")")
print(document2.debugDescription())
```

Outputs the title:
```
Document
└─ Paragraph
   └─ Image source: "/url" title: "title"
      └─ Text "foo"
```

Is this a version thing?

Thanks,
   Damian

### PR DESCRIPTION
**Pull request by Kyle-Ye**
Bug/issue #, if applicable: Close #67

## Summary

Add link title support for CommonMark

## Dependencies

None

## Testing

See new test case in `LinkTests.swift`

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: Tests/MarkdownTests/Inline Nodes/LinkTests.swift
Comment (Kyle-Ye) [inline Tests/MarkdownTests/Inline Nodes/LinkTests.swift:74]: Actually, I expect this to be
```suggestion
        XCTAssertEqual(linkWithoutTitle.title, nil)
```

But that's how `destination` and `title` is computed.

> `cmark_node_get_title` and `cmark_node_get_url` will use `cmark_chunk_to_cstr` which will produce an empty string for a nil input.

We can fix it by updating the getter logic on `Link.title`. The question is should we update the destination's getter accordingly? If so, it may break the downstream since it was a public symbol.

```
/// The link's title.
var title: String? {
    get {
        guard case let .link(_, title) = _data.raw.markup.data else {
            fatalError("\(self) markup wrapped unexpected \(_data.raw)")
        }
		if let t = title, t.isEmpty {
			return nil
		} else {
	        return title
		}
    }
	...
}
```

Comment (d-ronnqvist) [reply Tests/MarkdownTests/Inline Nodes/LinkTests.swift:74]: Alternatively we could check if the `cmark_node_get_title` value was empty in CommonMarkConverter.swift before passing the title to the `MarkupConversion` initializer

Comment (d-ronnqvist) [reply Tests/MarkdownTests/Inline Nodes/LinkTests.swift:74]: But I agree that I would also expect `nil` here

Comment (Kyle-Ye) [reply Tests/MarkdownTests/Inline Nodes/LinkTests.swift:74]: Both implementations are fine for me. The only question is should we sync such logic to `Link.destination`?

Comment (Kyle-Ye) [reply Tests/MarkdownTests/Inline Nodes/LinkTests.swift]: > Alternatively we could check if the `cmark_node_get_title` value was empty in CommonMarkConverter.swift before passing the title to the `MarkupConversion` initializer

+1 for this since we've already similar operations for `language`.

https://github.com/apple/swift-markdown/blob/07da205e1fc488d88609fea5da1ca227efc6c000/Sources/Markdown/Parser/CommonMarkConverter.swift#L329

Comment (Kyle-Ye) [reply Tests/MarkdownTests/Inline Nodes/LinkTests.swift]: Done with 2766399

**Comment by Kyle-Ye** [comment]
@swift-ci please test

**Review by d-ronnqvist — COMMENTED** [review]
This _seems_ good to me but I'm not familiar enough with the details to know the potential pitfalls.

**Comment by Kyle-Ye** [comment]
> This _seems_ good to me but I'm not familiar enough with the details to know the potential pitfalls.

Got it. Thanks for your review. Maybe we can wait for @QuietMisdreavus 's review suggestion.

**Comment by d-ronnqvist** [comment]
> > This _seems_ good to me but I'm not familiar enough with the details to know the potential pitfalls.
> 
> Got it. Thanks for your review. Maybe we can wait for @QuietMisdreavus 's review suggestion.

I would be fine with merging this and addressing any concerns from @QuietMisdreavus later, if there are any.

**Review by d-ronnqvist — APPROVED** [review]

**Comment by Kyle-Ye** [comment]
@swift-ci please test

**Comment by Kyle-Ye** [comment]
@swift-ci please test

**Comment by Kyle-Ye** [comment]
> > > This _seems_ good to me but I'm not familiar enough with the details to know the potential pitfalls.
> > 
> > 
> > Got it. Thanks for your review. Maybe we can wait for @QuietMisdreavus 's review suggestion.
> 
> I would be fine with merging this and addressing any concerns from @QuietMisdreavus later, if there are any.

Ping @QuietMisdreavus

**Comment by franklinsch** [comment]
It looks like this is breaking tests in Swift-DocC: https://ci.swift.org/job/oss-swift-package-amazon-linux-2/2319/consoleText

@Kyle-Ye Are you able to take a look?

**Comment by franklinsch** [comment]
I will revert this change to unblock Swift toolchains from being built and we can land this change back alongside the Swift-DocC changes once they're ready.

**Comment by Kyle-Ye** [comment]
> It looks like this is breaking tests in Swift-DocC: [ci.swift.org/job/oss-swift-package-amazon-linux-2/2319/consoleText](https://ci.swift.org/job/oss-swift-package-amazon-linux-2/2319/consoleText)
> 
> @Kyle-Ye Are you able to take a look?

The breaking is expected. Sorry I forgot to update the test case for swift-docc.

Here is the fix PR: https://github.com/apple/swift-docc/pull/708

**Pull request merged**
